File: src/kangla.Domain/Entities/WateringCommand.cs
using kangla.Domain.Interfaces;
using System.ComponentModel.DataAnnotations;

namespace kangla.Domain.Entities
{
    public class WateringCommand : IEntity
    {
        [Required]
        public int Id { get; set; }
        [Required]
        public int WateringDeviceId { get; set; }
        public WateringDevice WateringDevice { get; set; } = default!;
        [Required]
        public WateringCommandStatus Status { get; set; } = WateringCommandStatus.Pending;
        [Required]
        [Range(1, 60)]
        public int DurationSeconds { get; set; }
        [Required]
        public DateTime RequestedAtUtc { get; set; }
        [Required]
        public DateTime ExpiresAtUtc { get; set; }
        public DateTime? AcknowledgedAtUtc { get; set; }
        public DateTime? StartedAtUtc { get; set; }
        public DateTime? FinishedAtUtc { get; set; }
        [StringLength(200)]
        public string? FailureReason { get; set; }
        public int? WateringEventId { get; set; }
        public WateringEvent? WateringEvent { get; set; }
    }
}

File: src/kangla.client/src/app/watering-commands/watering-command.ts
export type WateringCommandStatus = 'pending' | 'acknowledged' | 'completed' | 'failed' | 'cancelled' | 'expired' | 'timedOut';
export type ActiveWateringCommandStatus = Extract<WateringCommandStatus, 'pending' | 'acknowledged'>;

export function isActiveWateringCommandStatus(
  status: WateringCommandStatus | null | undefined
): status is ActiveWateringCommandStatus {
  return status === 'pending' || status === 'acknowledged';
}

export function getActiveWateringCommandLabel(status: ActiveWateringCommandStatus): string {
  return status === 'pending' ? 'Watering queued' : 'Watering now';
}

export interface WateringCommand {
  id: number;
  deviceId: number;
  status: WateringCommandStatus;
  durationSeconds: number;
  requestedAtUtc: string;
  expiresAtUtc: string;
  acknowledgedAtUtc: string | null;
  startedAtUtc: string | null;
  finishedAtUtc: string | null;
  failureReason: string | null;
  wateringEventId: number | null;
}
